packages/mobile/src/fiatExchanges/BidaliScreen.tsx

```diff
@@ -0,0 +1,148 @@
+import colors from '@celo/react-components/styles/colors'
+import { CURRENCIES, CURRENCY_ENUM } from '@celo/utils'
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useEffect, useMemo, useRef, useState } from 'react'
+import { ActivityIndicator, StyleSheet } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { WebView, WebViewMessageEvent } from 'react-native-webview'
+import { useDispatch, useSelector } from 'react-redux'
+import { e164NumberSelector } from 'src/account/selectors'
+import { openUrl } from 'src/app/actions'
+import { bidaliPaymentRequested } from 'src/fiatExchanges/actions'
+import networkConfig from 'src/geth/networkConfig'
+import { celoTokenBalanceSelector } from 'src/goldToken/selectors'
+import i18n from 'src/i18n'
+import { emptyHeader } from 'src/navigator/Headers'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { TopBarTextButton } from 'src/navigator/TopBarButton'
+import { StackParamList } from 'src/navigator/types'
+import { stableTokenBalanceSelector } from 'src/stableToken/reducer'
+
+function useInitialJavaScript(
+  currency: CURRENCY_ENUM,
+  jsonBalances: string,
+  e164PhoneNumber: string | null
+) {
+  const [initialJavaScript, setInitialJavaScript] = useState<string | null>()
+  useEffect(() => {
+    if (initialJavaScript || !e164PhoneNumber) {
+      return
+    }
+
+    setInitialJavaScript(`
+      window.valora = {
```

Comment:
IIUC, Bidali is expecting all these properties to be injected into the global `window` object right? Then they will call `onPaymentRequest` or `openUrl`, which will take the inputs they've given us and writes it to the `window` objet for Valora to read from?

Then likewise, Bidali is expecting us to call `paymentSent` or `paymentCancelled` so that they can update their screens as well?

Given the uniqueness of this situation, I think some more comments explaining the interaction would be helpful.

Reply:
Yes I'll add a comment!